src/render/decorations/CHyprGroupBarDecoration.cpp

```diff
@@ -137,10 +141,14 @@ void CHyprGroupBarDecoration::draw(CMonitor* pMonitor, float a, const Vector2D&
             rect.y -= rect.height;
             rect.width = BARW * pMonitor->scale;
 
-            refreshGradients();
+            if (*PGRADIENTS == 1) {
+                if (m_tGradientActive.m_iTexID == 0) // TODO: find why texture doesn't exist
```

Comment:
like I said in original issue, despite texture being created previously on the create call they somehow don't exist here, being created 2 times instead of just 1

Reply:
try to gdb it maybe? I can try later too.

Reply:
on a closer looks seems like `glGenTextures(1, &m_iTexID);` gets run but the value of `m_iTexID` does not change

Reply:
thats bad

Reply:
output is 0

Reply:
output of the funtion? if gles logs any errors I believe it'd go to stdout/stderr. Idk what return codes there are tho

Reply:
using `glGetError()` I get an output of 0 which corresponds to `GL_NO_ERROR`
https://registry.khronos.org/OpenGL-Refpages/gl4/html/glGetError.xhtml
nothing is written to stdout/stderr

Reply:
ah, this. No clue then.

> some guy on a forum said it might be caused by it running asynchronously from the thread where the context was created, but I don't think that is the cause here

entire pipeline is single-threaded

Reply:
ok, seems like that was right
`glXGetCurrentContext()` is `0x0` in those 4 call while a different value everywhere else

Reply:
are you sure there are no threads, seems like everywhere that is called by events doesn't have an `eglGetCurrentContext()`

Reply:
only threads are for socket2 iirc. Main renderer is definitely single-threaded.

Reply:
this issue is now back since d2c3b23ace745d3f79926c71d4cb3c481a394841
the gradients are never getting rendered on main
adding the extra rendering in the main loop fixes it, like before